kubeflow/openmpi/assets.libsonnet

```diff
@@ -0,0 +1,128 @@
+{
+  all(params):: {
+    kind: "ConfigMap",
+    apiVersion: "v1",
+    metadata: {
+      name: "openmpi-assets",
+      labels: {
+        app: params.name,
+      },
+    },
+    data: {
+      "gen_hostfile.sh": |||
+        set -exv
+
+        # Download kubectl
+        apk add --update ca-certificates
+        apk add --update -t deps curl
+        curl -L https://storage.googleapis.com/kubernetes-release/release/v1.9.6/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl
+        chmod +x /usr/local/bin/kubectl
+
+        # Look up hosts and prepare hostfile for openmpi
+        hostfile=/kubeflow/openmpi/workdir/hostfile
+        until [ "$(wc -l < ${hostfile})" -eq "%(workers)d" ]; do
+          rm -f ${hostfile}
+          pod_names=$(kubectl -n %(namespace)s get pod --selector=role=worker --field-selector=status.phase=Running -o=jsonpath='{.items[*].metadata.name}')
```

Comment:
This seems a bit brittle. The selector seems underspecified; e.g. what if you have more than 1 job in a namespace?

Why not follow @inc0's suggestion in #535 of using  a stateful set? Then each worker would have a well defined hostname and we don't need to contact the APIServer to get the pod names because they are deterministic?

Alternatively, 

Why not create a simple go program to create the hostfile for the job and store it in a configmap?

The advantage of making this a go program is
- better testing
- better error handling
- better logging

Why not follow @inc0's suggestion in #535 of using  a stateful set? Then each worker would have a well defined hostname.

Reply:
The workers are already stateful sets and have a well defined hostname (in the form of "<APP>-worker-<INDEX>"), but I don't want to hard code the INDEX here. The selector also makes sure all the workers are in running state before generating the hostfile.

This PR doesn't support multiple jobs in the same namespace. Given that this PR is already pretty large, I'm going to address that in a follow-up PR.

In terms of embedding go program or other initialization scripts in the docker image, I agree that it'll make the code cleaner and easier to unit test, but then we would need to modify the docker images to embed those files. 

By embedding the scripts in ConfigMap, I am able to decouple the docker image and this kubeflow app. The goal is to deploy any docker images (say horovod) without understanding how kubeflow works internally.

Reply:
The other advantage is that I can update the scripts easily without rebuilding the docker images.